docker/api/Dockerfile

```
@@ -22,4 +22,5 @@ COPY requirements*.txt /home/kernelci/
 ARG REQUIREMENTS=requirements.txt
 RUN pip install --requirement ${REQUIREMENTS}
 
-CMD ["uvicorn", "api.main:app", "--host", "0.0.0.0", "--reload"]
+# Run migrations and start API server
+CMD pymongo-migrate migrate 20221014061654_set_timeout -u mongodb://db/kernelci -m ./migrations -v; uvicorn api.main:app --host 0.0.0.0 --reload
```

```suggestion
CMD pymongo-migrate migrate 20221014061654_set_timeout -u mongodb://db/kernelci -m ./migrations -v && uvicorn api.main:app --host 0.0.0.0 --reload
```
I think we shouldn't start service if db migrations failed.

---

Agreed.

---

```
pymongo-migrate migrate 20221014061654_set_timeout -u mongodb://db/kernelci -m ./migrations -v
```
That's quite a few hard-coded things that we can't just leave like that in the Dockerfile.  The list of migrations should be determined automatically.

---

I have already dropped the target file name from the command:
```
pymongo-migrate migrate -u mongodb://db/kernelci -m ./migrations -v
```
The default migration directory to be automatically detected by the package is `pymongo_migrations`.
I rather used a generic name for the directory i.e. `migrations`.
I think we can use build args to provide db host, db name, and migrations directory to avoid hard cording.

---

Added env variables for DB host, DB name, and directory of migration files to make them dynamic.